src/sentry/api/serializers/models/group.py

```diff
@@ -722,22 +771,56 @@ def __init__(
         self.stats_period_start = stats_period_start
         self.stats_period_end = stats_period_end
         self.matching_event_id = matching_event_id
+        self.has_dynamic_issue_counts = has_dynamic_issue_counts
 
-    def query_tsdb(self, group_ids, query_params):
+    def _get_seen_stats(self, item_list, user):
+        partial_execute_seen_stats_query = functools.partial(
+            self._execute_seen_stats_query,
+            item_list=item_list,
+            environment_ids=self.environment_ids,
+            start=self.start,
+            end=self.end,
+        )
+        time_range_result = partial_execute_seen_stats_query()
+        if self.has_dynamic_issue_counts:
+            filtered_result = (
+                partial_execute_seen_stats_query(conditions=self.conditions)
+                if self.conditions
+                else None
+            )
+            lifetime_result = (
+                partial_execute_seen_stats_query(start=None, end=None)
+                if self.start or self.end
+                else None
+            )
+            for item in item_list:
+                time_range_result[item].update({"filtered": filtered_result.get(item)})
+                time_range_result[item].update({"lifetime": lifetime_result.get(item)})
```

Comment:
```suggestion
                time_range_result[item].update({
                    "filtered": filtered_result.get(item),
                    "lifetime": lifetime_result.get(item),
                })
```
You could combine these updates.